feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -548,13 +573,29 @@ class PreviewViewModel @AssistedInject constructor(
                                 lastBlinkTimeStamp = System.currentTimeMillis()
                             ) ?: old
                         }
-                    }, contentResolver, imageCaptureUri, ignoreUri).savedUri
+                    }, contentResolver, finalImageUri, ignoreUri).savedUri
+                },
+                onSuccess = { savedUri ->
+                    onImageCapture(ImageCaptureEvent.ImageSaved(savedUri), uriIndex)
                 },
-                onSuccess = { savedUri -> onImageCapture(ImageCaptureEvent.ImageSaved(savedUri)) },
                 onFailure = { exception ->
-                    onImageCapture(ImageCaptureEvent.ImageCaptureError(exception))
+                    onImageCapture(ImageCaptureEvent.ImageCaptureError(exception), uriIndex)
                 }
             )
+            incrementExternalMultipleImageCaptureModeUriIndexIfNeeded()
+        }
+    }
+
+    private fun incrementExternalMultipleImageCaptureModeUriIndexIfNeeded() {
+        if (previewUiState.value is PreviewUiState.Ready &&
+            (previewUiState.value as PreviewUiState.Ready).previewMode
+                is PreviewMode.ExternalMultipleImageCaptureMode &&
+            !(
+                (previewUiState.value as PreviewUiState.Ready).previewMode
+                    as PreviewMode.ExternalMultipleImageCaptureMode
+                ).imageCaptureUris.isNullOrEmpty()
+        ) {
+            externalUriIndex++
```

Comment:
nit: Can make this more concise
```
private fun incrementExternalMultipleImageCaptureModeUriIndexIfNeeded() {
    ((previewUiState.value as? PreviewUiState.Ready)
        ?.previewMode as? PreviewMode.ExternalMultipleImageCaptureMode)?.let {
        if (!it.imageCaptureUris.isNullOrEmpty()) {
            externalUriIndex++
        }
    }
}
```

Or you could just do something like initialize `externalUriIndex` to `-1` if the mode isn't `ExternalMultipleImageCaptureMode`. Then you can do
```
if (externalUriIndex >= 0) {
    externalUriIndex++
}
```